tests/unit/src/Internal/Payment/State/InitialStateTest.php

```diff
@@ -132,52 +105,68 @@ public function test_process( Exception $exception = null ) {
 		$this->sut->set_context( $this->mock_context );
 
-		// There's a single call to get the order ID.
-		$this->mock_context->expects( $this->once() )
+		// Mock get order id calls.
+		$this->mock_context->expects( $this->exactly( 2 ) )
 			->method( 'get_order_id' )
-			->willReturn( $order_id );
+			->willReturn( 1 );
 
 		// Verify that the context is populated.
 		$this->sut->expects( $this->once() )->method( 'populate_context_from_request' )->with( $mock_request );
 		$this->sut->expects( $this->once() )->method( 'populate_context_from_order' );
 
-		// Arrange intent creation.
-		if ( $exception ) {
-			$this->mock_payment_request_service->expects( $this->once() )
-				->method( 'create_intent' )
-				->with( $this->mock_context )
-				->willThrowException( $exception );
-		} else {
-			$this->mock_payment_request_service->expects( $this->once() )
-				->method( 'create_intent' )
-				->with( $this->mock_context )
-				->willReturn( $mock_intent );
-
-			// Assert order update.
-			$this->mock_order_service->expects( $this->once() )
-				->method( 'update_order_from_successful_intent' )
-				->with( $order_id, $mock_intent, $this->mock_context );
-		}
-
-		// Arrange the final state.
-		$state_class      = $exception ? SystemErrorState::class : CompletedState::class;
-		$mock_final_state = $this->createMock( $state_class );
-		$this->mock_state_factory->expects( $this->once() )
+		// Since the original create_state method is mocked, we have to manually set context.
+		$this->mock_state_factory->expects( $this->exactly( 3 ) )
 			->method( 'create_state' )
-			->with( $state_class, $this->mock_context )
-			->willReturn( $mock_final_state );
-
-		// Act: Process.
-		$result = $this->sut->process( $mock_request );
-
+			->withConsecutive(
+				[ VerifiedState::class, $this->mock_context ],
+				[ ProcessedState::class, $this->mock_context ],
+				[ CompletedState::class, $this->mock_context ]
+			)
+			->willReturnOnConsecutiveCalls(
+				( function () {
+					$intent = WC_Helper_Intention::create_intention();
+
+					$mock_payment_request_service = $this->createMock( PaymentRequestService::class );
+					$mock_payment_request_service->expects( $this->once() )
+						->method( 'create_intent' )
+						->with( $this->mock_context )
+						->willReturn( $intent );
+
+					$this->mock_context->expects( $this->exactly( 2 ) )
+						->method( 'get_intent' )
+						->willReturn( $intent );
+
+					$verified_state = new VerifiedState( $this->mock_state_factory, $mock_payment_request_service );
+					$verified_state->set_context( $this->mock_context );
+
+					return $verified_state;
+				} )(),
+				( function () {
+					$this->mock_order_service->expects( $this->once() )
+						->method( 'update_order_from_successful_intent' )
+						->with( $this->mock_context->get_order_id(), $this->mock_context->get_intent(), $this->mock_context );
+					$processed_state = new ProcessedState( $this->mock_state_factory, $this->mock_order_service );
```

Comment:
This is not testing the `InitialState`, it's testing the whole process, which is not needed, and over-complicates this test. Let's just return a mock state from the first `create_state` call, assert that the right method is called, and be done with it. Otherwise, once we add all the needed logic to the next state, this test will grow out of control.

Reply:
Addressed here: https://github.com/Automattic/woocommerce-payments/pull/7471/commits/776c96fa45b2f0da36dd2483b2d60a645bf04abf